src/runtime/server/middleware/session/index.ts

```diff
@@ -17,10 +19,41 @@ const safeSetCookie = (event: H3Event, name: string, value: string) => setCookie
   sameSite: useRuntimeConfig().session.session.cookieSameSite as SameSiteOptions
 })
 
-export declare interface Session {
-  id: string
-  createdAt: Date
-  [key: string]: any
+const checkSessionExpirationTime = (session: Session, sessionOptions: SessionOptions) => {
+  const sessionExpiryInSeconds = sessionOptions.expiryInSeconds
+  if (sessionExpiryInSeconds === null) {
+    return
+  }
+
+  const now = dayjs()
+  if (now.diff(dayjs(session.createdAt), 'seconds') > sessionExpiryInSeconds) {
+    throw new SessionExpired()
+  }
+}
+
+const checkSessionIp = async (event: H3Event, session: Session, sessionOptions: SessionOptions) => {
```

Comment:
Since you're already separated this logic into the function, lets move it to the rest of the logic related to ip-pinning inside `ipPinning.ts`. Reason: other developers working with the code wouldn't need to know/scroll/read ipPinning code if they work on another feature